### PR TITLE
[3.2.x] Fixed test_request_lifecycle_signals_dispatched_with_thread_sensitive with asgiref 3.5.1+.

### PR DESCRIPTION
**Pull request by felixxm**
See [comment](https://code.djangoproject.com/ticket/33795#comment:2) and [logs](https://djangoci.com/view/Main/job/django-3.2/83/).

### CODE REVIEW

File: tests/asgi/tests.py [hunk 1]
```diff
@@ -235,11 +235,8 @@ def __call__(self, **kwargs):
...
         await communicator.wait()
 
-        # At this point, AsyncToSync does not have a current executor. Thus
-        # SyncToAsync falls-back to .single_thread_executor.
-        target_thread = next(iter(SyncToAsync.single_thread_executor._threads))
+        # AsyncToSync should have executed the signals in the same thread.
         request_started_thread, request_finished_thread = signal_handler.threads
-        self.assertEqual(request_started_thread, target_thread)
-        self.assertEqual(request_finished_thread, target_thread)
+        self.assertEqual(request_started_thread, request_finished_thread)
```
Comment (carltongibson) [inline tests/asgi/tests.py:240]: This is probably fine... We loose the **this thread** check but...

Comment (felixxm) [reply tests/asgi/tests.py:240]: I tried to backport the necessary part of 36fa071d6ebd18a61c4d7f1b5c9d17106134bd44 to make it works with all supported versions of `asgiref`. We don't have this check in 4.0+ branches :thinking:

Comment (carltongibson) [reply tests/asgi/tests.py:240]: It's likely fine. 🤔

Let me have a play in the debugger tomorrow.

**Comment by carltongibson** [comment]
So this diff maintains the test for the specific thread: 


```diff

-from asgiref.sync import SyncToAsync
+from asgiref.sync import AsyncToSync, SyncToAsync
 from asgiref.testing import ApplicationCommunicator
 
 from django.contrib.staticfiles.handlers import ASGIStaticFilesHandler
@@ -235,9 +235,8 @@ class ASGITest(SimpleTestCase):
         # Give response.close() time to finish.
         await communicator.wait()
 
-        # At this point, AsyncToSync does not have a current executor. Thus
-        # SyncToAsync falls-back to .single_thread_executor.
-        target_thread = next(iter(SyncToAsync.single_thread_executor._threads))
+        executor = AsyncToSync.loop_thread_executors[asyncio.get_running_loop()]
+        target_thread = executor._work_thread
         request_started_thread, request_finished_thread = signal_handler.threads
         self.assertEqual(request_started_thread, target_thread)
         self.assertEqual(request_finished_thread, target_thread)
```

Same passes if applied to `main`. 

I'm inclined to say it's too dependent on the internals of `asgiref.sync` — It was useful solving the original issue, but I don't see that we need to maintain that still. 

So... 👍

**Review by carltongibson — APPROVED** [review]

**Comment by felixxm** [comment]
@carltongibson Thanks for checking :+1:

**Pull request merged**